boards: Add "ram" and "flash" properties for a number of boards

pfalcon opened the pull request:
Currently, it seems that only boards with smaller amount of RAM
get a value for "ram" property, and very few boards have "flash"
defined. However, to test bigger sample applications, it's helpful
to have these properties universally defined.

Signed-off-by: Paul Sokolovsky <paul.sokolovsky@linaro.org>

pfalcon committed:
boards: Add "ram" and "flash" properties for a number of boards

Currently, it seems that only boards with smaller amount of RAM
get a value for "ram" property, and very few boards have "flash"
defined. However, to test bigger sample applications, it's helpful
to have these properties universally defined.

Signed-off-by: Paul Sokolovsky <paul.sokolovsky@linaro.org>

pfalcon commented:
This popped up while working on a test for https://github.com/zephyrproject-rtos/zephyr/pull/5985 - despite specifying `min_ram` and `min_flash` (for a biggish mbedTLS application), I was getting linker errors. Turned out, many board yaml files miss the needed properties.

galak commented:
How about making sanitycheck dt aware so these come from there instead.

nashif commented:
> How about making sanitycheck dt aware so these come from there instead.

yes, i will start looking at this for 1.12, i think there is an issue for this.

pfalcon commented:
> yes, i will start looking at this for 1.12, i think there is an issue for this.

@nashif, thanks!

pfalcon commented:
Otherwise, the CI of this gets affected by https://github.com/zephyrproject-rtos/zephyr/issues/6616, so I'm going to include fix for it here too.

pfalcon committed:
tests: bluetooth: shell: Set min_flash to 145K

With recent master, this sample started to overflow ROM area of
arduino_101, which has 144K of it:

Memory region         Used Size  Region Size  %age Used
             ROM:      149420 B       144 KB    101.33%
             RAM:       26228 B        52 KB     49.26%
        IDT_LIST:         564 B         2 KB     27.54%
        MMU_LIST:          0 GB         1 KB      0.00%

Signed-off-by: Paul Sokolovsky <paul.sokolovsky@linaro.org>

MaureenHelm reviewed (approved):
We should go ahead and merge this to unblock @pfalcon and because it's an existing feature being just applied to more boards. After @nashif figures out how to get this info from dt instead, we can remove the redundant info from the board yamls.

galak commented on boards/x86/minnowboard/minnowboard.yaml:
```diff
@@ -4,3 +4,4 @@ type: mcu
 arch: x86
 toolchain:
   - zephyr
+ram: 256
```
why no flash size?

nashif replied:
we never flash anything on this board. This is loaded via EFI and can be in GBs

galak reviewed (commented):
Why don't we just update arduino_101 and the test for now.  Seems silly to add all these, just to remove them later.

galak reviewed (changes requested):
Increasing min_flash to limit arduino_101 isn't the right answer, someone needs to look at why 101 is so big and fix it since its testing features

pfalcon commented:
> Why don't we just update arduino_101 and the test for now.

Because not just arduino_101 is affected, but many (all?) boards for which these basic properties aren't specified.

> Seems silly to add all these, just to remove them later.

Well, that's called "development" and "refactoring" ;-). As pointed by @MaureenHelm, these things represent existing feature and can be added *now* (ahem, could have been added before). Where's taking it from DTS is just a modal idea.

galak commented:
Than how about we update sanitycheck to require these properties as part of this PR

andrewboie commented:
I think it's worth mentioning that we already have Kconfigs which express this information because the linker scripts require this data to set up the binary correctly.

There might be some variation in how this is expressed in Kconfig across arches, but I'd really like to see this sort of info pulled from one source, rather than maintaining it in a few different places. If DTS ends up being the canonical source that's fine w/me.

galak commented:
> I think it's worth mentioning that we already have Kconfigs which express this information because the linker scripts require this data to set up the binary correctly.
> 
> There might be some variation in how this is expressed in Kconfig across arches, but I'd really like to see this sort of info pulled from one source, rather than maintaining it in a few different places. If DTS ends up being the canonical source that's fine w/me.

DTS will end up being the source, that should be the case for the linker scripts that use this info.  We need to cleanup the issue across arch'es around what SRAM, FLASH, MEMORY, ICCM, DCCM, etc mean and are getting used for.

nashif commented:
> There might be some variation in how this is expressed in Kconfig across arches, but I'd really like to see this sort of info pulled from one source, rather than maintaining it in a few different places. If DTS ends up being the canonical source that's fine w/me.

Right, we want to pull all of this from DTS now that most boards/archs are supporting DTS. That was  a stop-gap.
